pkg/transparentproxy/iptables/parameters/parameters.go

```diff
@@ -54,20 +54,25 @@ func (p *Parameter) Build(verbose bool) string {
 	var parameters []string
 	for _, parameter := range p.parameters {
 		if parameter != nil {
-			parameters = append(parameters, parameter.Build(verbose))
+			parameters = append(parameters, parameter.Build(verbose)...)
 		}
 	}
 
 	// Some parameters for flags like "--wait" or "--wait-interval" require an
 	// equal sign to be set, so "--wait 5" is invalid and should be "--wait=5"
 	// to work. If the `Parameter` object has a `connector` property and only
 	// one value, we will use it when joining the flag with the value.
-	connector := " "
 	if p.connector != "" && len(parameters) == 1 {
-		connector = p.connector
+		lastElem := result[len(result)-1]
+		resultWithoutLastElem := result[: len(result)-1 : len(result)-1] // full slice expression to protect before append overwrites
+
+		return append(
```

Comment:
maybe `slices.Concat` makes this easier?

Reply:
Not in this case as this is an exception when a flag needs to have a value assigned with `=` (`--wait=5`)